greenpithumb/greenpithumb.py

```diff
@@ -15,22 +15,27 @@
 import moisture_sensor
 import pi_io
 import poller
+import pump
 import record_processor
+import sleep_windows
 import temperature_sensor
 import wiring_config_parser
 
 logger = logging.getLogger(__name__)
 
 
-def make_sensor_pollers(poll_interval, wiring_config, record_queue):
+def make_sensor_pollers(poll_interval, wiring_config, record_queue,
+                        sleep_windows):
     logger.info('creating sensor pollers (poll interval=%d")', poll_interval)
+    utc_clock = clock.Clock()
     local_clock = clock.LocalClock()
     # The MCP3008 spec and Adafruit library use different naming for the
     # Raspberry Pi GPIO pins, so we translate as follows:
     # * CLK -> CLK
     # * CS/SHDN -> CS
     # * DOUT -> MISO
     # * DIN -> MOSI
+    raspberry_pi_io = pi_io.IO(GPIO)
```

Comment:
Suggest just defining this inline since it's only used once, like

```python
water_pump = pump.Pump(pi_io.IO(GPIO), utc_clock, wiring_config.gpio_pins.pump)
```